static/spoken/templates/testimonial/mediaform.html

```diff
@@ -34,5 +41,24 @@ <h3>Are you sure you want to delete the testimonial?</h3>
                 </p>
             {% endif%}
         </div>
-    </div>
+        <video class='d-none' id="st_video">
+            <source src="" />
+        </video>
+    <script>
+        var media_element_container = document.getElementById('media_element');
+        var hidden_video_container = document.getElementById('st_video');
+        var submit_btn_container = document.getElementById('submit_btn');
+        media_element_container.onchange = function () {
+            hidden_video_container.src = URL.createObjectURL(this.files[0]);
+            hidden_video_container.load();
+        };
+        hidden_video_container.onloadedmetadata = function() {
+            // To change the allowed duration of the testimonial, please manipulate the variable below(seconds). 
```

Comment:
Checking the duration of the video on client end.
1. To change the allowed duration of the testimonial, the value of 
`var allowed_duration` can be changed (in seconds).
2. Limitations:
 - System assumes that clients browser supports modern JavaScript ( Like: Firefox / Chrome/ Safari)
 - Client is trusted person who will not remove the function to allow the desired video to be uploaded.
3. Benefits of the client side check:
- Video doesn't need to be stored in server.
- Faster check for client.